crates/bevy_ui/src/widget/text.rs

```diff
@@ -1,49 +1,204 @@
+    TextSpanAccess, TextStyle, TextWriter, YAxisOrientation,
 };
+use bevy_transform::components::Transform;
 use bevy_utils::{tracing::error, Entry};
 use taffy::style::AvailableSpace;
 
-/// Text system flags
+/// UI text system flags.
 ///
 /// Used internally by [`measure_text_system`] and [`text_system`] to schedule text for processing.
 #[derive(Component, Debug, Clone, Reflect)]
 #[reflect(Component, Default, Debug)]
-pub struct TextFlags {
-    /// If set a new measure function for the text node will be created
-    needs_new_measure_func: bool,
-    /// If set the text will be recomputed
+pub struct TextNodeFlags {
+    /// If set then a new measure function for the text node will be created.
+    needs_measure_fn: bool,
+    /// If set then the text will be recomputed.
     needs_recompute: bool,
 }
 
-impl Default for TextFlags {
+impl Default for TextNodeFlags {
     fn default() -> Self {
         Self {
-            needs_new_measure_func: true,
+            needs_measure_fn: true,
             needs_recompute: true,
         }
     }
 }
 
+/// The top-level UI text component.
+///
+/// Adding `TextNEW` to an entity will pull in required components for setting up a UI text node.
+///
+/// The string in this component is the first 'text span' in a hierarchy of text spans that are collected into
+/// a [`TextBlock`]. See [`TextSpan`] for the component used by children of entities with `TextNEW`.
+///
+/// Note that [`Transform`] on this entity is managed automatically by the UI layout system.
+///
+/*
+```
+# use bevy_asset::Handle;
+# use bevy_color::Color;
+# use bevy_color::palettes::basic::BLUE;
+# use bevy_ecs::World;
+# use bevy_text::{Font, JustifyText, TextBlock, TextStyle};
+# use bevy_ui::Text;
+#
+# let font_handle: Handle<Font> = Default::default();
+# let mut world = World::default();
+#
+// Basic usage.
+world.spawn(Text::new("hello world!"));
+
+// With non-default style.
+world.spawn((
+    Text::new("hello world!"),
+    TextStyle {
+        font: font_handle.clone().into(),
+        font_size: 60.0,
+        color: BLUE.into(),
+    }
+));
+
+// With text justification.
+world.spawn((
+    Text::new("hello world\nand bevy!"),
+    TextBlock::new_with_justify(JustifyText::Center)
+));
+```
+*/
+#[derive(Component, Debug, Default, Clone, Deref, DerefMut, Reflect)]
+#[reflect(Component, Default, Debug)]
+#[require(
+    TextBlock,
+    TextStyle,
+    TextNodeFlags,
+    Node,
+    Style, // TODO: Remove when Node uses required components.
+    ContentSize, // TODO: Remove when Node uses required components.
+    FocusPolicy, // TODO: Remove when Node uses required components.
+    ZIndex, // TODO: Remove when Node uses required components.
+    BackgroundColor, // TODO: Remove when Node uses required components.
+    Visibility, // TODO: Remove when Node uses required components.
+    Transform // TODO: Remove when Node uses required components.
+)]
+pub struct TextNEW(pub String);
```

Comment:
I really dislike how the first span is a different type. I'd prefer if it just required the first text span to be a child too.
Could we do something like put the `require` on a marker component and have a constructor that returns a tuple `(Marker, TextSpan)`?

Reply:
The original proposal had a universal span component but cart wanted to consolidate: https://github.com/bevyengine/bevy/discussions/15014#discussioncomment-10574459

Reply:
> If we aren't already doing it, we could make text.0 for the default empty string behave as "no span", so the first child of Text::default() -> TextSpan is at index 0.

I disagree with this, it creates ambiguity and conceptual branching.

Reply:
> I disagree with this, it creates ambiguity and conceptual branching. Note that it's common to set a default-empty span and then later write text to it.

I don't have strong opinions on that. Just pitching an alternative path that allows `Text::default()->TextSpan` to behave as if the first TextSpan child is actually the first span. I do agree that it makes the "empty Text to non-empty" change harder to reason about, as it would behave differently than an empty child TextSpan.